src/item.cpp

```diff
@@ -2033,6 +2033,13 @@ nc_color item::color_in_inventory() const
                 ret = c_pink;
             } else if( !u->studied_all_recipes( *type ) ) { // Book can't improve skill anymore, but has more recipes: yellow
                 ret = c_yellow;
+            } else if ( type->can_use( "MA_MANUAL" ) ) { // Martial arts book support
+                // TODO: replace this terrible hack to rely on the item name matching the style name, it's terrible.
+                // (also change equivalent code in iuse.cpp's ma_manual)
+                const matype_id style_to_learn( "style_" + typeId().substr( 7 ) );
```

Comment:
Please don't multiply hacks in the code. It would be better if you add a new variable to `islot_book` (say, `martial_art`) and isolate the hack inside `Item_factory::load( islot_book &slot, ...)` or `Item_factory::load_book()`. 
This way you could prepare it for moving to JSON and thus getting rid of the hack.